### PR TITLE
Update HarfBuzzSharp

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

New environment variables were introduced for each supported platform to only reference the native assets of the current platform

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

### CODE REVIEW

File: build/HarfBuzzSharp.props [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
-    <PackageReference Condition="'$(IncludeWasmSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.WebAssembly" Version="2.8.2.3" />
+    <PackageReference Include="HarfBuzzSharp" Version="7.3.0" />
+    <PackageReference Condition="'$(IncludeWindowsSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.Win32" Version="7.3.0" />
+    <PackageReference Condition="'$(IncludeLinuxSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.Linux" Version="7.3.0" />
+    <PackageReference Condition="'$(IncludeMacSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.macOS" Version="7.3.0" />
+    <PackageReference Condition="'$(IncludeWasmSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.WebAssembly" Version="7.3.0" />
+    <PackageReference Condition="'$(IncludeAndroidSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.Android" Version="7.3.0" />
+    <PackageReference Condition="'$(IncludeIOSSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.iOS" Version="7.3.0" />
+    <PackageReference Condition="'$(IncludeTizenSkia)' == 'true'" Include="HarfBuzzSharp.NativeAssets.Tizen" Version="7.3.0" />
```
Comment (maxkatz6) [inline build/HarfBuzzSharp.props:10]: Everything but WASM and Linux should still be included automatically. You can see from dependencies page here: https://www.nuget.org/packages/HarfBuzzSharp

.NETStandard 2.0 has Win32 and macOS
And .NET 6 has mobile platforms as well.

File: src/Android/Avalonia.Android/Avalonia.Android.csproj [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <MSBuildEnableWorkloadResolver>true</MSBuildEnableWorkloadResolver>
     <DebugType>portable</DebugType>
     <AndroidResgenNamespace>Avalonia.Android.Internal</AndroidResgenNamespace>
+    <IncludeAndroidSkia>true</IncludeAndroidSkia>
```
Comment (maxkatz6) [inline src/Android/Avalonia.Android/Avalonia.Android.csproj:9]: I think these properties work only when build/HarfBuzzSharp.props is referenced directly in the project.

File: build/HarfBuzzSharp.props (hunk 1 above)
Comment (Gillibald) [reply build/HarfBuzzSharp.props:10]: Hmm so it looks like everything is referenced automatically

Comment (Gillibald) [reply build/HarfBuzzSharp.props:10]: https://github.com/mono/SkiaSharp/blob/main/binding/IncludeNativeAssets.HarfBuzzSharp.targets

Comment (maxkatz6) [reply build/HarfBuzzSharp.props]: Yes, everything automatically, except linux and non-blazor browser. But we already cover these platforms manually.

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**